src/Bicep.Cli/Commands/BuildParamCommand.cs

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Linq;
+using System.Threading.Tasks;
+using Bicep.Cli.Arguments;
+using Bicep.Cli.Logging;
+using Bicep.Cli.Services;
+using Bicep.Core;
+using Bicep.Core.Parsing;
+using Bicep.Core.Syntax;
+using Bicep.Core.FileSystem;
+using Microsoft.Extensions.Logging;
+using Bicep.Core.Diagnostics;
+
+namespace Bicep.Cli.Commands
+{
+    public class BuildParamCommand
+    {
+        private readonly ILogger logger;
+        private readonly IDiagnosticLogger diagnosticLogger;
+        private readonly InvocationContext invocationContext;
+        private readonly CompilationService compilationService;
+        private readonly CompilationWriter writer;
+
+        public BuildParamCommand(
+            ILogger logger,
+            IDiagnosticLogger diagnosticLogger,
+            InvocationContext invocationContext,
+            CompilationService compilationService,
+            CompilationWriter writer)
+        {
+            this.logger = logger;
+            this.diagnosticLogger = diagnosticLogger;
+            this.invocationContext = invocationContext;
+            this.compilationService = compilationService;
+            this.writer = writer;
+        }
+
+        public async Task<int> RunAsync(BuildArguments args)
+        {
+            var inputPath = PathHelper.ResolvePath(args.InputFile);
+            var inputUri = PathHelper.FilePathToFileUrl(inputPath);
+
+            if (PathHelper.HasExtension(inputUri, LanguageConstants.ParamsFileExtension))
+            {
+                var parser = new Parser(inputPath);
+                var program = parser.Program();
+                var usingDeclarations = program.Children.OfType<UsingDeclarationSyntax>();
+                var paramPath = TryGetUsingPath(usingDeclarations.SingleOrDefault(), out var getUsingPathFailureBuilder);
```

Comment:
It feels like now might be the right time to create the compilation / semantic model class and put this logic there. That might also make combining the sets of changes more straightforward (see my other comment)